src/com/tutorial/main/Handler.java

```diff
@@ -1,51 +0,0 @@
-package com.tutorial.main;
-
-import java.awt.Graphics;
-import java.util.ArrayList;
-import java.util.LinkedList;
-
-public class Handler {
-	
-	LinkedList<GameObject> object = new LinkedList<GameObject>();
-	LinkedList<GameObject> toAdd = new LinkedList<GameObject>();
-	
-	public void tick(){
-		// Will declaring this inside of the loop make it clear after each call to tick()?
-		ArrayList<Integer> toRemoveIndex = new ArrayList<Integer>();
-		
-		// Update Objects
-		for(GameObject tempObject : object){
-			tempObject.tick();
-			
-			// Save index of array objects to be removed
-			if (tempObject.removed) {
-				toRemoveIndex.add(object.indexOf(tempObject));
-			}
-		}
-	
-		// Remove marked objects
-		for (int index = toRemoveIndex.size() - 1; index >= 0; index--) {
-			this.removeObject(object.get(toRemoveIndex.get(index)));
-		}
```

Comment:
Yeah that's one thing you'll have to pay attention to with java. The same underlying object can be referenced as many different types.
For example, if you have a class definition:
````
public final class MyActivity extends Activity implements Runnable, View.OnClickListener {...}
````
Then create a new `MyActivity`. The following assignments would all work and the resulting objects would all be the same:
````
MyActivity a = new MyActivity();
Activity b = a;
Runnable c = a;
View.OnClickListener d = a;
````